extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java

```diff
@@ -0,0 +1,464 @@
+            throw new IllegalArgumentException("fenodes", feNodes);
+        }
+        List<String> nodes = Arrays.asList(feNodes.split(","));
+        Collections.shuffle(nodes);
+        return nodes.get(0).trim();
+    }
+
+    /**
+     * get a valid URI to connect Doris FE.
+     * @param cfg configuration of request
+     * @param logger {@link Logger}
+     * @return uri string
+     * @throws IllegalArgumentException throw when configuration is illegal
+     */
+    @VisibleForTesting
+    static String getUriStr(Settings cfg, Logger logger) throws IllegalArgumentException {
+        String[] identifier = parseIdentifier(cfg.getProperty(DORIS_TABLE_IDENTIFIER), logger);
+        return "http://" +
+                choiceFe(cfg.getProperty(DORIS_FENODES), logger) + API_PREFIX +
+                "/" + identifier[0] +
+                "/" + identifier[1] +
+                "/";
+    }
+
+    /**
+     * discover Doris table schema from Doris FE.
+     * @param cfg configuration of request
+     * @param logger slf4j logger
+     * @return Doris table schema
+     * @throws DorisException throw when discover failed
+     */
+    public static Schema discoverSchema(Settings cfg, Logger logger)
+            throws DorisException {
+        logger.trace("Finding schema.");
+        HttpGet httpGet = new HttpGet(getUriStr(cfg, logger) + SCHEMA);
+        String response = send(cfg, httpGet, logger);
+        logger.debug("Find schema response is '{}'.", response);
+        return feResponseToSchema(response, logger);
+    }
+
+    /**
+     * translate Doris FE response to inner {@link Schema} struct.
+     * @param response Doris FE response
+     * @param logger {@link Logger}
+     * @return inner {@link Schema} struct
+     * @throws DorisException throw when translate failed
+     */
+    @VisibleForTesting
+    public static Schema feResponseToSchema(String response, Logger logger) throws DorisException {
+        logger.trace("Parse response '{}' to schema.", response);
+        ObjectMapper mapper = new ObjectMapper();
+        Schema schema;
+        try {
+            schema = mapper.readValue(response, Schema.class);
+        } catch (JsonParseException e) {
+            String errMsg = "Doris FE's response is not a json. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        } catch (JsonMappingException e) {
+            String errMsg = "Doris FE's response cannot map to schema. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        } catch (IOException e) {
+            String errMsg = "Parse Doris FE's response to json failed. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        }
+
+        if (schema == null) {
+            logger.error(SHOULD_NOT_HAPPEN_MESSAGE);
+            throw new ShouldNeverHappenException();
```

Comment:
```suggestion
            throw new NotFoundException();
```

Reply:
this is a fatal error, 'shoud never happen' is meaningful